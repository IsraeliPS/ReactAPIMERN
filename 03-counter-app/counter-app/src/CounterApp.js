import React from 'react';
import PropTypes from 'prop-types';

function CounterApp({value}) {
    return (
        <>
         <h1>CounterApp</h1>   
         <h2>{value}</h2>
         <button >+1</button>
        </>
    )
}

CounterApp.protoTypes={
    value:PropTypes.number
}
export default CounterApp