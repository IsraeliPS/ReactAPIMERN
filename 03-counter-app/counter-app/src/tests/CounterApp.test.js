import React from 'react'
import { shallow } from 'enzyme'
import '@testing-library/jest-dom'

import CounterApp from '../CounterApp'
import '../setupTest'

describe('Pruebas en Elemento CounterApp', () => {
  let wrapper = shallow(<CounterApp value={123} />)

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={123} />)
  })

  test('debe mostrar <CounterApp /> correctamente y sus valore', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('Debe mostrar el valor por defecto', () => {
    const valor = '100'
    const wrapper = shallow(<CounterApp value={valor} />)
    const cantidadEnviada = wrapper.find('h2').text()
    expect(cantidadEnviada).toBe(valor)
  })

  test('Debe de incrementar con el botón +1', () => {
    wrapper.find('button').at(0).simulate('click')
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe('124')
  })

  test('Debe de decrementar con el botón -1', () => {
    wrapper.find('button').at(2).simulate('click')
    const counterText = wrapper.find('h2').text()
    expect(counterText).toBe('122')
  })

  test('Debe de colocar el valor por defecto con el boton reset', () => {
    const valor = 100
    const wrapper = shallow(<CounterApp value={valor} />)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    const counterText = wrapper.find('h2').text()
    expect(valor).toBe(100)
  })
})
