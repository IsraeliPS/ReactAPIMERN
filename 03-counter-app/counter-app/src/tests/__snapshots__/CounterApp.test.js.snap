// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Elemento CounterApp debe mostrar <CounterApp /> correctamente y sus valore 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    123
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
